edgexfoundry/device-rfid-llrp-go
# fix: update snap packaging

- Add support for defining provision watcher directory
- Add auto-scan script to snap
- Add support in auto-configure script/command for providing Consul token
- Update snap base to core 20
- Update documentation
- Update metadata and versioning
- Add support for "snap set" of device-specific settings
- Update install hook
- Update make section in snapcraft.yaml
- Other minor updates

Fix https://github.com/edgexfoundry/device-rfid-llrp-go/issues/55

See also: https://github.com/edgexfoundry/device-rfid-llrp-go/pull/52 for further information.

There is an issue with setting the subnet information using "snap set", which is unrelated to this PR - see edgexfoundry/app-functions-sdk-go#1043

Signed-off-by: Siggi Skulason <siggi.skulason@canonical.com>







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-rfid-llrp-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  <link to docs PR>

## Testing Instructions


Install snap and run device discovery with
```
sudo snap remove edgexfoundry
sudo snap remove edgex-device-rfid-llrp
sudo snap install edgexfoundry --edge

snapcraft try
sudo snap try prime

sudo snap connect edgexfoundry:edgex-secretstore-token edgex-device-rfid-llrp:edgex-secretstore-token
sudo snap start edgex-device-rfid-llrp.device-rfid-llrp 
sudo snap connect edgex-device-rfid-llrp:network-control 
sudo snap connect edgex-device-rfid-llrp:network-observe 

CONSUL_TOKEN=$(sudo cat /var/snap/edgexfoundry/current/secrets/consul-acl-token/bootstrap_token.json | jq ".SecretID" | tr -d '"') 
echo $CONSUL_TOKEN

sudo edgex-device-rfid-llrp.auto-configure $CONSUL_TOKEN

```

I have verified that this detects my RFID reader.


## New Dependency Instructions (If applicable)
